File: leetcode/0260.go
package leetcode

func singleNumber(nums []int) []int {
	tmp := 0
	for _, num := range nums {
		tmp ^= num
	}
	tmp &= -tmp
	ans := []int{0, 0}
	for _, num := range nums {
		if num&tmp == 0 {
			ans[0] ^= num
		} else {
			ans[1] ^= num
		}
	}
	return ans
}

File: leetcode/0260.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        unsigned int tmp = 0;
        for (auto& num : nums) tmp ^= num;
        tmp &= -tmp;
        auto ans = vector<int>{0, 0};
        for (auto& num : nums)
            if (tmp & num)
                ans[0] ^= num;
            else
                ans[1] ^= num;
        return ans;
    }
};